src/problems/PrimordialChem/test_primordial_chem.cpp

```diff
@@ -6,6 +6,10 @@
 /// \file test_primordial_chem.cpp
 /// \brief Defines a test problem for primordial chemistry (microphysics).
 ///
+#ifdef __HIP_DEVICE_COMPILE__
+#undef _GLIBCXX_ASSERTIONS
+#endif
+
```

Comment:
This is an alternative workaround. Unfortunately, this preprocessor magic needs to be placed at the top of every *.cpp file that has compile errors.

Reply:
For some reason, this did not work.